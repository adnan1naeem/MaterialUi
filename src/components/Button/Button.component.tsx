/**
 * Created by M.GhazanfarAli on 12/27/17.
 */

import * as React from 'react';
import IButton from '@blueeast/bluerain-ui-interfaces/Components/Button';
import { Button } from 'material-ui';


/**
 * The props of Button Component
 * @param {Node} props.children Label to be passed as child.
 * @param {Object} props.style style object for button's style.
 * @param {enum} props.color color prop of type enum. Default color is 'primary'.
 * @param {function} props.onClick callback function fired when button is pressed.
 * @param {boolean} props.disabled if true, renders a disabled button.
 * @param {string} props.href href prop to move to a link.
 * @param {boolean} props.fullWidth if true, button is generated with 100% width of the cnotainer.
 * @param {Node} props.iconLeft icon prop to show icon on the left side of label.
 * @param {Node} props.iconRight icon prop to show icon on the right side of label.
 * @param {boolean} props.fab if true, fab button is generated.
 * @param {boolean} props.small if true, small button is generated (small also works when fab is true).
 * @param {string} props.className string name of styling css class.
 */

 export interface ButtonMUI {
        color?: 'default' | 'accent' | 'primary' | 'contrast' | 'inherit',
        dense?: boolean,
        classes?: object,
        disableRipple?: boolean,
        disableFocusRipple?: boolean,
        component?: string | undefined,
        raised?: boolean,
 }

const BlueRainButton = (props: IButton & ButtonMUI) => {
    if(props.iconLeft){
    return (
        <Button
        raised={true || props.raised}
        color={ props.color ? (props.color ==='default'|| props.color ==='accent'|| props.color ==='primary'|| props.color ==='contrast'? props.color: 'default') : 'default'} 
        style={props.fullWidth ? (props.style, {width:'100%'}): props.style} 
        disabled={props.disabled}
        href={props.href}
        fab={props.fab}
        mini={props.small}
        dense={props.small}
        onClick={props.onClick}
        {...props}
        >
        {props.iconLeft} {props.children}
        </Button>
    );
    }
    else if(props.iconRight){
        return (
            <Button
            raised={true || props.raised}
            color={ props.color ? (props.color==='default'|| props.color==='accent'|| props.color==='primary'|| props.color==='contrast'? props.color: 'default') : 'default'} 
            style={props.fullWidth? (props.style, {width:'100%'}): props.style} 
            disabled={props.disabled}
            href={props.href}
            fab={props.fab}
            mini={props.small}
            dense={props.small}
            onClick={props.onClick}
            {...props}
            >
            {props.children} {props.iconRight}
            </Button>
        );
        }
    return (
        <Button
        raised={true || props.raised}
        color={ props.color? (props.color==='default'|| props.color==='accent'|| props.color==='primary'|| props.color==='contrast'? props.color: 'default') : 'default'} 
        style={props.fullWidth ? (props.style, {width:'100%'}): props.style} 
        disabled={props.disabled}
        href={props.href}
        fab={props.fab}
        mini={props.small}
        dense={props.small}
        onClick={props.onClick}
        {...props}
        >
        {props.children}
        </Button>
    );

}

export default BlueRainButton;