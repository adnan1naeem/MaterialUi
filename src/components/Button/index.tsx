// export { BlueRainButton as Button } from  './Button.component';