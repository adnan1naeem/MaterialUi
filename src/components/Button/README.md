# Button

Buttons allow users to take actions, and make choices, with a single tap.

## Compatibility

| 🌏 Web | 🖥 Electron | 📱 React Native |
| :----: | :---------: | :-------------: |
| ✔️      | ✖            | ✖              |

## Screenshots

| 🌏 Web | 🖥 Electron | 📱 React Native |
| :---: | :--------: | :------------: |
| ![web image](./screenshots/Button.png) |    TBD   | TBD |

## Universal Props

| Name | Type | Default | Description |
|:-----|:-----|:--------|:------------|
| style | Object | | style object for button. |
| children | Node | | Content of the button component. |
| color | enum:'default' ,'inherit' ,'primary','secondary' | 'default' | Color of button component.|
| onClick | func | () => {} | Callback function fired when button is clicked. |
| disabled | bool |  | If true, renders a disabled button. |
| href | string | | href prop to move to a link. |
| fullWidth | bool |  | If true, button is generated with 100% width of the cnotainer. |
| iconLeft | Node |  | icon prop to show icon on the left side of label. |
| iconRight | Node | | icon prop to show icon on the right side of label. |
| small | bool |  | If true, renders a small sized button. |
| large | bool |  | If true, renders a large sized button.|
| variant | enum |'text' , 'flat' , 'outlined' , 'contained' , 'raised' , 'fab' , 'extendedFab'  | If true, renders a large sized button.|

## Specific Library Props

| Name | Type | Default | Description |
|:-----|:-----|:--------|:------------|
| tag | func, string | button | tag prop to generate component with any other component and not with default 'button'. |
| size | enum:'small','medium','large' |  | Useful to set size of button component. |

## How to use

```JavaScript
import React from 'react';
import Button from  'bluerain-plugin-bootstrap-ui/Button';

const buttonSample = () => (
       <Button fullWidth color='primary'>Click me!</Button>
);

export default buttonSample;
```