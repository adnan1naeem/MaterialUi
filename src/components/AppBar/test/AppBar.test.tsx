import * as React from 'react';
import Button from 'material-ui';
import { AppBar } from '../index';
import renderer from 'react-test-renderer';
import  Adapter from 'enzyme-adapter-react-16';
import { configure, mount, shallow, } from 'enzyme';

configure({ adapter: new Adapter() });

test('AppBar component with position prop', () => {
	const appBar = shallow(
      <AppBar position="static"></AppBar>
  );
	expect(appBar.props().position).toBeUndefined();
	expect(appBar).toMatchSnapshot();
});

test('AppBar component with style', () => {
	const appBar = shallow(
      <AppBar style={{ 'background-color':'green' }}></AppBar>
  );
	expect(appBar.props().style).toBeUndefined();
	expect(appBar).toMatchSnapshot();
});

test('AppBar component with children', () => {
	const appBar = shallow(
      <AppBar style={{ 'background-color':'green' }} left={"Hello"}>Hello</AppBar>
  );
	expect(appBar.props().left).toBeUndefined();
	expect(appBar).toMatchSnapshot();
});