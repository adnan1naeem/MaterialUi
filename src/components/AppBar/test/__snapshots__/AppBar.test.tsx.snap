// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar component with children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    left="Hello"
    style={
        Object {
            "background-color": "green",
          }
    }
>
    Hello
</Unknown>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`AppBar component with position prop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    position="static"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`AppBar component with style 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    style={
        Object {
            "background-color": "green",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
